source/unified-test-format/unified-test-format.rst

```diff
@@ -1798,6 +1857,155 @@ Use a ``listIndexes`` command to check whether the index exists. Note that it is
+  - ``time``: the number of (floating-point) seconds since the Unix epoch
+    when the error was encountered.
+
+- ``storeFailuresAsEntity``: if specified, the runner MUST capture failures
+  arising during sub-operation execution and append a document with failure
+  information to the array stored in the specified entity.
+  
+  If this option is specified, the test runner MUST check the existence and
+  the type of the entity with the specified name before executing the loop.
+  If the entity does not exist, the test runner MUST create it with the type
+  of BSON array. If the entity exists and is of type BSON array, the
+  test runner MUST do nothing. If the entity exists and is of a different type,
+  the test runner MUST raise an error.
+  
+  If this option is specified and ``storeErrorsAsEntity`` is not, errors
+  MUST also be captured and appended to the array.
+
+  Documents appended to the array MUST contain the following fields:
+  
+  - ``error``: the textual description of the failure encountered.
+  - ``time``: the number of (floating-point) seconds since the Unix epoch
+    when the failure was encountered.
+
+- ``storeSuccessesAsEntity``: if specfied, the runner MUST keep track of
+  the number of sub-operations that completed successfully, and store
+  that number in the specified entity. For example, if the loop contains
+  two sub-operations, and they complete successfully, each loop execution
+  would increment the number of successes by two.
+  
+  If the entity of the specified name already exists, the test runner
+  MUST raise an error.
+
+- ``storeIterationsAsEntity``: if specfied, the runner MUST keep track of
+  the number of iterations of the loop performed, and store that number
+  in the specified entity.
+  
+  If the entity of the specified name already exists, the test runner
+  MUST raise an error.
+
+A *failure* is when the result or outcome of an operation executed by the
+test runner differs from its expected outcome. For example, an ``expectResult``
+assertion failing to match a BSON document or an ``expectError`` assertion
+failing to match an error message would be considered a failure. An *error*
+is any other type of error raised by the test runner. For example, an
+unsupported operation or inability to resolve an entity name would be
+considered an error.
+
+This specification permits the test runner to report some failures as errors
+and some errors as failures. When the test runner stores errors and
+failures as entities it MAY classify conditions as errors and failures in
+the same way as it would when used in the driver's test suite.
+This includes reporting all errors as failures or all failures as errors.
+
+If the test runner does not distinguish errors and failures in its reporting,
+it MAY report both conditions under either category, but it MUST report
+any given condition in at most one category.
+
+The same entity name MAY be used for both ``storeErrorsAsEntity``
```

Comment:
I'm not sure about this. This requires special handling by a driver to not blindly overwrite an entity value, and it's an extra effort done to undo treating errors and failures differently. I'd vote for either treating errors and as separate categories and omitting this option for writing them to the same entity, or to treat failures and errors exactly the same, which removes the need for this special handling.

Reply:
I removed this provision.

Reply:
The changes to test format on DAT side mandate distinct entity names for errors and failures. The shared name was something I thought might be useful in the future, but since we have clarified why the errors and the failures must be separate, the use case for combining them into one entity doesn't appear to be strong anymore.